sdk/eventhub/Azure.Messaging.EventHubs/src/PartitionContext.cs

```diff
@@ -12,22 +16,91 @@ namespace Azure.Messaging.EventHubs
     ///
     public class PartitionContext
     {
+        /// <summary>
+        ///   The name of the Event Hub this context is associated with.
+        /// </summary>
+        ///
+        public string EventHubName { get; }
+
         /// <summary>
         ///   The identifier of the Event Hub partition this context is associated with.
         /// </summary>
         ///
         public string PartitionId { get; }
 
+        /// <summary>
+        ///   The <see cref="TransportConsumer" /> for this context to use as the source for information.
+        /// </summary>
+        ///
+        private WeakReference<TransportConsumer> SourceConsumer { get; }
+
+        /// <summary>
+        ///   A set of information about the last enqueued event of a partition, as observed by the <see cref="EventHubConsumerClient" />
+        ///   associated with this context as events are received from the Event Hubs service.  This is only available if the consumer was
+        ///   created with <see cref="EventHubConsumerClientOptions.TrackLastEnqueuedEventInformation" /> set.
+        /// </summary>
+        ///
+        /// <remarks>
+        ///   When information about the partition's last enqueued event is being tracked, each event received from the Event Hubs
+        ///   service will carry metadata about the partition that it otherwise would not. This results in a small amount of
+        ///   additional network bandwidth consumption that is generally a favorable trade-off when considered
+        ///   against periodically making requests for partition properties using an Event Hub client.
+        /// </remarks>
+        ///
+        /// <exception cref="EventHubsClientClosedException">Occurs when the <see cref="EventHubConsumerClient" /> needed to read this information is no longer available.</exception>
+        /// <exception cref="InvalidOperationException">Occurs when this method is invoked without <see cref="EventHubConsumerClientOptions.TrackLastEnqueuedEventInformation" /> set.</exception>
+        ///
+        public virtual LastEnqueuedEventProperties ReadLastEnqueuedEventInformation()
+        {
+            var consumer = default(TransportConsumer);
+
+            if ((SourceConsumer?.TryGetTarget(out consumer) == false) || (consumer == null))
+            {
+                // If the consumer instance was not available, treat it as a closed instance for
+                // messaging consistency.
+
+                Argument.AssertNotClosed(true, Resources.ConsumerClientNeededForThisInformation);
```

Comment:
Yeah, looks a bit weird, but its as-intended.   The caller doesn't know how that is retrieved, so telling them "the consumer client" or similar would be confusing.  As would saying the context was closed, since it can't be.

So, this is a generic bit of text that basically would say "The client needed for the information has been closed and cannot perform the operation."